### PR TITLE
BLUE-1484: Move from createdBy (deprecated) to the new createdBys filter in CommandDefinition search endpoint

### PR DESCRIPTION
**Pull request by VivienLG**

### CODE REVIEW

File: helper/general_objects/schemas.go [hunk 1]
```diff
@@ -310,7 +310,10 @@ func DefinitionAttributeSchema(excludeTypes []string, excludeFields []string, fo
 
 		// Enum only
 		"options": {
-			Type:        schema.TypeMap,
+			Type: schema.TypeMap,
+			Elem: &schema.Schema{
+				Type: schema.TypeString,
+			},
```
Comment (VivienLG) [inline helper/general_objects/schemas.go:316]: Not related to this ticket, but all enum options parameters should be a map of String.

File: services/commands/command_definitions/schemas.go [hunk 2]
```diff
@@ -164,8 +164,11 @@ var dataSourceFilterSchema = map[string]*schema.Schema{
 		Optional:    true,
 		Description: "Search by name or description",
 	},
-	"created_by": {
-		Type:         schema.TypeString,
+	"created_bys": {
+		Type: schema.TypeList,
+		Elem: &schema.Schema{
+			Type: schema.TypeString,
+		},
```
Comment (VivienLG) [inline services/commands/command_definitions/schemas.go:171]: Previous createdBy has been replaced by createdBys, to be plateform consistent

**Comment by VivienLG** [comment]
I don't know why, but when I run the main.tf file under commandDefinition, I always get the following error:
```bash
│ 6 errors occurred:
│       * resource leanspace_metrics: appliedArgumentId: Field name may only contain lowercase alphanumeric characters & underscores.
│       * resource leanspace_activity_definitions: appliedArgumentId: Field name may only contain lowercase alphanumeric characters & underscores.
│       * resource leanspace_command_definitions: appliedArgumentId: Field name may only contain lowercase alphanumeric characters & underscores.
│       * data source leanspace_activity_definitions: appliedArgumentId: Field name may only contain lowercase alphanumeric characters & underscores.
│       * data source leanspace_command_definitions: appliedArgumentId: Field name may only contain lowercase alphanumeric characters & underscores.
│       * data source leanspace_metrics: appliedArgumentId: Field name may only contain lowercase alphanumeric characters & underscores.
```
And we don't have any appliedArgumentId fields in the code. So pretty strange.

**Review by elodietinland — APPROVED** [review]

File: services/commands/command_definitions/schemas.go
Comment (jerome-leanspace) [inline services/commands/command_definitions/schemas.go:1]: Can you also modify examples/data_sources/leanspace_command_definitions.tf and test/commands/command_definitions/main.tf to reflect this change?

**Comment by jerome-leanspace** [comment]
> ```shell
> appliedArgumentId
> ```

Could it be a change because of Java 17 in commands?

**Review by kbudkaLeanspace — APPROVED** [review]

**Comment by VivienLG** [comment]
> I don't know why, but when I run the main.tf file under commandDefinition, I always get the following error:
> 
> ```shell
> │ 6 errors occurred:
> │       * resource leanspace_metrics: appliedArgumentId: Field name may only contain lowercase alphanumeric characters & underscores.
> │       * resource leanspace_activity_definitions: appliedArgumentId: Field name may only contain lowercase alphanumeric characters & underscores.
> │       * resource leanspace_command_definitions: appliedArgumentId: Field name may only contain lowercase alphanumeric characters & underscores.
> │       * data source leanspace_activity_definitions: appliedArgumentId: Field name may only contain lowercase alphanumeric characters & underscores.
> │       * data source leanspace_command_definitions: appliedArgumentId: Field name may only contain lowercase alphanumeric characters & underscores.
> │       * data source leanspace_metrics: appliedArgumentId: Field name may only contain lowercase alphanumeric characters & underscores.
> ```
> 
> And we don't have any appliedArgumentId fields in the code. So pretty strange.

Fixed

**Review by jerome-leanspace — APPROVED** [review]

**Pull request merged**